samples/drivers/rtc/src/main.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2024, Muhammad Waleed Badar
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/rtc.h>
+#include <zephyr/sys/util.h>
+
+const struct device *const rtc = DEVICE_DT_GET(DT_ALIAS(rtc));
+
+#ifdef SET_DATE_TIME
+static int set_date_time(const struct device *rtc)
+{
+	int ret = 0;
+	struct rtc_time tm = {.tm_year = 2024 - 1900,
+			      .tm_mon = 7 - 1,
+			      .tm_mday = 28,
+			      .tm_hour = 3,
+			      .tm_min = 51,
+			      .tm_sec = 0};
+
+	ret = rtc_set_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot write date time: %d\n", ret);
+		return ret;
+	}
+	return ret;
+}
+#endif
+
+static int get_date_time(const struct device *rtc)
+{
+	int ret = 0;
+	struct rtc_time tm;
+
+	ret = rtc_get_time(rtc, &tm);
+	if (ret < 0) {
+		printk("Cannot read date time: %d\n", ret);
+		return ret;
+	}
+
+	printk("Current RTC time: %04d-%02d-%02d %02d:%02d:%02d\n", tm.tm_year + 1900,
+	       tm.tm_mon + 1, tm.tm_mday, tm.tm_hour, tm.tm_min, tm.tm_sec);
+
+	return ret;
+}
+
+int main(void)
+{
+	/* Check if the RTC is ready */
+	if (!device_is_ready(rtc)) {
+		printk("Device is not ready\n");
+		return 0;
+	}
+
+#ifdef SET_DATE_TIME
+	set_date_time(rtc);
+#endif
```

Comment:
For this sample, it should probably always set the time :) so `SET_DATE_TIME` should be removed. If you really want it, the define should be a Kconfig like `CONFIG_TEST_SET_DATE_TIME`

See:
https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/drivers/rtc/rtc_api/Kconfig
https://github.com/zephyrproject-rtos/zephyr/blob/c50777a8431157bd4ab3a2c2e2a37e8151ae07da/tests/drivers/rtc/rtc_api/src/test_alarm.c#L19

Reply:
removed `#ifdef SET_DATE_TIME`